effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -1454,15 +1467,15 @@ trait TyperOps extends ContextOps { self: Context =>
 
   private [typer] def bindingCapabilities[R](binder: source.Tree, caps: List[symbols.BlockParam])(f: => R): R = {
     bindCapabilities(binder, caps)
-    capabilityScope = BindSome(binder, caps.map { c => c.tpe.asInterfaceType -> c }.toMap, capabilityScope)
+    capabilityScope = BindSome(binder, caps.map { c => c.tpe.get.asInterfaceType -> c }.toMap, capabilityScope)
```

Comment:
Do we know by construction that there always will be a type?

Reply:
Yes, we do either get the `caps` argument from `Context.freshCapabilityFor` where, by construction, a type is always given or from line 219 where we also ensure a block type is given (or abort if not). So we do not get a runtime error further down the line when making changes, I added the usual `Context.abort("type required...")`. Is this sufficient?